File: app/src/main/java/com/bryan/platform/model/entity/Comment.java
package com.bryan.platform.model.entity;

import androidx.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * 评论实体数据模型（Android 客户端用）
 * 对应后端 Moment 实体中的内嵌 Comment 文档
 */
public class Comment implements Serializable {

    @SerializedName("id")
    private String id;

    @SerializedName("content")
    private String content;

    @SerializedName("authorId")
    private Long authorId;

    @SerializedName("authorName")
    private String authorName;

    @SerializedName("createdAt")
    private String createdAt;

    public Comment() {
    }

    public Comment(String id,
                   String content,
                   Long authorId,
                   String authorName,
                   String createdAt) {
        this.id = id;
        this.content = content;
        this.authorId = authorId;
        this.authorName = authorName;
        this.createdAt = createdAt;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public Long getAuthorId() {
        return authorId;
    }

    public void setAuthorId(Long authorId) {
        this.authorId = authorId;
    }

    public String getAuthorName() {
        return authorName;
    }

    public void setAuthorName(String authorName) {
        this.authorName = authorName;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }

    @NonNull
    @Override
    public String toString() {
        return "Comment{" +
                "id='" + id + '\'' +
                ", content='" + content + '\'' +
                ", authorId=" + authorId +
                ", authorName='" + authorName + '\'' +
                ", createdAt='" + createdAt + '\'' +
                '}';
    }
}

File: app/src/main/java/com/bryan/platform/model/entity/Comment.kt
package com.bryan.platform.model.entity

import com.google.gson.annotations.SerializedName
import java.io.Serializable

/**
 * 评论实体数据模型，用于 Android 客户端。
 * 对应后端 Moment 实体中的内嵌 Comment 文档。
 *
 * @property id 评论的唯一ID (通常由应用或服务器生成)。
 * @property content 评论的文本内容。
 * @property authorId 发表评论的用户的ID。
 * @property authorName 发表评论的用户的显示名称。
 * @property createdAt 评论的创建时间 (以ISO-8601格式字符串表示，如 "2025-07-15T10:00:00")。
 */
data class Comment(
    @SerializedName("id")
    val id: String?,

    @SerializedName("content")
    val content: String?,

    @SerializedName("authorId")
    val authorId: Long?,

    @SerializedName("authorName")
    val authorName: String?,

    @SerializedName("createdAt")
    val createdAt: String?
) : Serializable